library/src/meta/figure.rs

```diff
@@ -77,21 +338,119 @@ impl Show for FigureElem {
 
 impl Count for FigureElem {
     fn update(&self) -> Option<CounterUpdate> {
+        // If the figure is numbered, step the counter by one.
+        // This steps the `counter(figure)` which is global to all numbered figures.
```

Comment:
does it only step the global one? what's with unnumbered headings that have a kind?

Reply:
I gave it a try and it has an impact, if a figure has a kind and a numbering it gets counted just fine, if it has a kind but no numbering it does not get counted in either counters from what I can tell, which is the behaviour I was expecting. So it is needed as-is. If you disable counting all together then no counter work.

It behaves in exactly how I was expecting it to which is reassuring